crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -1207,6 +1208,35 @@ impl<TX: DbTx + 'static, N: NodeTypesForProvider> BlockReader for DatabaseProvid
 
         Ok(None)
     }
+
+    fn block_body(&self, id: BlockHashOrNumber) -> ProviderResult<Option<Self::BlockBody>> {
+        if let Some(number) = self.convert_hash_or_number(id)? {
+            if self.block_body_indices(number)?.is_some() {
+                // If the body indices are not found, this means that the transactions either do not
+                // exist in the database yet, or they do exit but are not indexed.
+                // If they exist but are not indexed, we don't have enough
+                // information to return the block anyways, so we return `None`.
+                let Some(transactions) = self.transactions_by_block(number.into())? else {
+                    return Ok(None)
+                };
+
+                // We need a header to pass to read_block_bodies, but we only care about the body
+                if let Some(header) = self.header_by_number(number)? {
+                    let body = self
+                        .storage
+                        .reader()
+                        .read_block_bodies(self, vec![(&header, transactions)])?
```

Comment:
oh I totally forgot that we actually need the header as input here... sigh

then this kinda defeats the purpose, I cant think of a workaround here.

hmm, then this feature no longer makes sense unfortunately...

very sorry about that